cmd/test/test.go

```diff
@@ -165,60 +165,67 @@ func newTestCommand() *cobra.Command {
 			}
 
 			if !runner.Quiet || exitCode != 0 {
-				if runner.Output == OutputAppstudio {
-					// The appstudio format is unknown to Conftest so we handle it ourselves
+				for _, outputAndFormat := range outputFormats {
+					parts := strings.SplitN(outputAndFormat, "=", 2)
 
-					if resultsErr != nil {
-						return appstudioErrorHandler(runner.NoFail, "running test", resultsErr)
+					format := parts[0]
+					if format == "" {
+						// default to stdout
+						format = output.OutputStandard
 					}
 
-					report := appstudioReport(results, runner.Namespace)
-					reportOutput, err := json.Marshal(report)
-					if err != nil {
-						return appstudioErrorHandler(runner.NoFail, "output results", err)
+					outputFilePath := ""
+					if len(parts) == 2 {
+						outputFilePath = parts[1]
 					}
+					if format == OutputAppstudio {
+						// The appstudio format is unknown to Conftest so we handle it ourselves
 
-					if outputFilePath != "" {
-						err := os.WriteFile(outputFilePath, reportOutput, 0600)
+						if resultsErr != nil {
+							return appstudioErrorHandler(runner.NoFail, "running test", resultsErr)
+						}
+
+						report := appstudioReport(results, runner.Namespace)
+						reportOutput, err := json.Marshal(report)
 						if err != nil {
-							return fmt.Errorf("creating output file: %w", err)
+							return appstudioErrorHandler(runner.NoFail, "output results", err)
 						}
-					}
-					fmt.Fprintln(cmd.OutOrStdout(), string(reportOutput)+"\n")
 
-				} else {
-					// Conftest handles the output
+						if outputFilePath != "" {
+							err := os.WriteFile(outputFilePath, reportOutput, 0600)
+							if err != nil {
+								return fmt.Errorf("creating output file: %w", err)
+							}
+						} else {
+							fmt.Fprintln(cmd.OutOrStdout(), string(reportOutput)+"\n")
+						}
 
-					if resultsErr != nil {
-						return fmt.Errorf("running test: %w", resultsErr)
-					}
+					} else {
+						// Conftest handles the output
 
-					var outputFile *os.File
-					if outputFilePath != "" {
-						outputFile, err = os.Create(outputFilePath)
-						if err != nil {
-							return fmt.Errorf("creating output file: %w", err)
+						if resultsErr != nil {
+							return fmt.Errorf("running test: %w", resultsErr)
 						}
-						defer outputFile.Close()
-					}
 
-					outputter := output.Get(runner.Output, output.Options{
-						NoColor:            runner.NoColor,
-						SuppressExceptions: runner.SuppressExceptions,
-						Tracing:            runner.Trace,
-						JUnitHideMessage:   viper.GetBool("junit-hide-message"),
-						File:               outputFile,
-					})
-					if err := outputter.Output(results); err != nil {
-						return fmt.Errorf("output results: %w", err)
-					}
+						var outputFile *os.File
+						if outputFilePath != "" {
+							outputFile, err = os.Create(outputFilePath)
+							if err != nil {
+								return fmt.Errorf("creating output file: %w", err)
```

Comment:
Maybe this will be better for debugging:

```suggestion
								return fmt.Errorf("creating output file %s: %w", outputFilePath, err)
```